### ISSUE TITLE
[bug] Shell API open command opens trunctated URL on Windows if URL contains ampersand

### ISSUE THREAD

**Issue by aghster** [tauri-apps/tauri#6473]
### Describe the bug

When using the `open` command of the Shell API to open a URL in the default browser and the URL contains multiple query params joined by `&`, the browser instead opens a truncated URL that only comprises the characters up to the first ampersand. 

### Reproduction

Trying to open `https://tauri.app/?a=1&b=2` in the default browser via

```ts
import { open } from '@tauri-apps/api/shell';
await open('https://tauri.app/?a=1&b=2');
```

opens `https://tauri.app/?a=1` instead.

### Expected behavior

I would expect that the original URL `https://tauri.app/?a=1&b=2` is opened.

### Platform and versions

```text
Environment
  › OS: Windows 10.0.22621 X64
  › Webview2: 110.0.1587.69
  › MSVC: 
      - Visual Studio Build Tools 2019
      - Visual Studio Build Tools 2022
  › Node.js: 18.13.0
  › npm: 8.19.3
  › pnpm: Not installed!
  › yarn: Not installed!
  › rustup: 1.25.1
  › rustc: 1.66.0
  › cargo: 1.66.0
  › Rust toolchain: stable-x86_64-pc-windows-msvc 

Packages
  › @tauri-apps/cli [NPM]: 1.2.3
  › @tauri-apps/api [NPM]: 1.2.0
  › tauri [RUST]: 1.2.4,
  › tauri-build [RUST]: 1.2.1,
  › tao [RUST]: 0.15.8,
  › wry [RUST]: 0.23.4,

App
  › build-type: bundle
  › CSP: unset
  › distDir: ../build
  › devPath: http://localhost:5173/
  › framework: Svelte
  › bundler: Vite

App directory structure
  ├─ .git
  ├─ .svelte-kit
  ├─ build
  ├─ enigo
  ├─ node_modules
  ├─ src
  ├─ src-tauri
  └─ static
```

### Additional context

As it seems, this is due to a [bug](https://github.com/Byron/open-rs/issues/67) in `open-rs` that has recently been [fixed](https://github.com/Byron/open-rs/pull/68). So probably, updating the dependency will fix this issue. However, the source contains some kind of [caveat](https://github.com/tauri-apps/tauri/blob/1fa729b218a2e0af57524a3b5f300f8f11ca12c9/core/tauri/src/scope/shell.rs#L317-L318) wrt how arguments are handled by this dependency.

**Comment by aghster** [comment]
I just found https://github.com/tauri-apps/tauri/pull/6441 which seems to address the same issue. However, I do not think that using `open 3.2.0` will help here, as the bug was fixed in `open 4.0.0` only. But `open 4.0.0` requires '1.62' as the minimal Rust version which is why Tauri's dependency on `open` shall currently not be bumped to `4.0.0`. So this remains a Tauri issue. 

Would it be possible to correctly escape the path the be opened automatically within Tauri? Is there some workaround?

**Comment by aghster** [comment]
There _is_ a workauround. I finally found out how to properly escape the `&` on Windows, i.e. by replacing it with `\^&`. The following correctly opens `https://tauri.app/?a=1&b=2` with all URL parameters in the default browser.

```ts
import { open } from '@tauri-apps/api/shell';
await open('https://tauri.app/?a=1&b=2'.replaceAll("&", "\^&"));
```

So, now there is a workaround, this issue could be closed. But I suggest including the workaround directly into Tauri, since one would not expect (and it could easily go unnoticed) that `open` silently omits all URL parameters except the first.
